ratis-examples/src/main/java/org/apache/ratis/examples/debug/server/Server.java

```diff
@@ -0,0 +1,81 @@
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ratis.examples.debug.server;
+
+import org.apache.ratis.conf.RaftProperties;
+import org.apache.ratis.examples.common.StateMachine;
+import org.apache.ratis.examples.common.Common;
+import org.apache.ratis.grpc.GrpcConfigKeys;
+import org.apache.ratis.protocol.RaftPeer;
+import org.apache.ratis.server.RaftServer;
+import org.apache.ratis.server.RaftServerConfigKeys;
+import org.apache.ratis.util.NetUtils;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.Collections;
+
+/**
+ * Simplest Ratis server, use a simple state machine {@link StateMachine}
+ * which maintain a counter across multi server.
+ *
+ * Run this application three times with three different parameter set-up a
+ * ratis cluster which maintain a counter value replicated in each server memory
+ */
+public final class Server {
+
+  private Server(){
+  }
+
+  public static void main(String[] args) throws IOException {
+    if (args.length < 1) {
+      System.err.println("Ratis sever address should be provided in the style <ip:port>");
+      System.exit(1);
+    }
+
+    //find current peer object based on application parameter
+    RaftPeer currentPeer =
+        Common.PEERS.stream()
+            .filter(raftPeer -> raftPeer.getAddress().equals(args[0]))
+            .findFirst().get();
+
+    //create a property object
+    RaftProperties properties = new RaftProperties();
+
+    //set the storage directory (different for each peer) in RaftProperty object
+    File raftStorageDir = new File("./" + currentPeer.getId().toString());
+    RaftServerConfigKeys.setStorageDir(properties, Collections.singletonList(raftStorageDir));
+
+    //set the port which server listen to in RaftProperty object
+    final int port = NetUtils.createSocketAddr(currentPeer.getAddress()).getPort();
+    GrpcConfigKeys.Server.setPort(properties, port);
+
+    //create the counter state machine which hold the counter value
+    StateMachine stateMachine = new StateMachine();
+
+    //create and start the Raft server
+    RaftServer server = RaftServer.newBuilder()
+        .setGroup(Common.RAFT_GROUP)
+        .setProperties(properties)
+        .setServerId(currentPeer.getId())
+        .setStateMachine(stateMachine)
+        .build();
+    server.start();
```

Comment:
Let's refactor the CounterServer so that we can reuse the code.

Reply:
done